google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/ProxyService.java

```diff
@@ -67,8 +70,10 @@ protected final void addServices(ApiService... services) throws ApiException {
   // Tries to stop all dependent services and sets this service into the FAILED state.
   protected final void onPermanentError(CheckedApiException error) {
     if (stoppedOrFailed.getAndSet(true)) return;
-    for (ApiService service : services) {
-      service.stopAsync();
+    try {
+      ApiServiceUtils.stopAsync(services);
+    } catch (Throwable t) {
+      LOGGER.atFine().withCause(t).log("Exception in underlying service shutdown.");
     }
     handlePermanentError(error);
```

Comment:
We should also likely handle exception from `handlePermanentError(error)` that was the line that actually caused issue for us.

Reply:
Good point. Fixed.